File: src/main/java/com/wafflestudio/draft/repository/UserRepository.java
package com.wafflestudio.draft.repository;

import com.wafflestudio.draft.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);

    Optional<User> findByUsername(String username);

    boolean existsUserByUsername(String username);
}

File: src/main/java/com/wafflestudio/draft/repository/UserRepository.kt
package com.wafflestudio.draft.repository

import com.wafflestudio.draft.model.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User?, Long?> {
    fun findByEmail(email: String?): Optional<User?>
    fun findByUsername(username: String?): Optional<User?>?
}